common/lib/api-client/middleware/request.js

```diff
@@ -0,0 +1,39 @@
+const { get } = require('lodash')
+
+const redisStore = require('../../../../config/redis-store')()
+const models = require('../models')
+
+module.exports = function requestMiddleware(expiry = 60) {
+  return {
+    name: 'axios-request',
+    req: async function req(payload) {
+      const { req, jsonApi } = payload
+      const pathname = new URL(req.url).pathname
+      const searchString = Object.keys(req.params).length
+        ? `?${new URLSearchParams(req.params).toString()}`
+        : ''
+      const key = `cache:${req.method}.${pathname}${searchString}`
+
+      if (!get(models, `${req.model}.options.cache`)) {
+        return jsonApi.axios(req)
+      }
+
+      const result = await redisStore.client
+        .getAsync(key)
+        .then(response => JSON.parse(response))
+
+      if (result) {
+        return { data: result }
```

Comment:
should this also be a returning a Promise? then the return is always a Promise - same as `jsonApi.axios(req)`?